[MAINTENANCE] Add typing stubs to `requirements-types.txt`

Kilo59 opened the pull request:
Adding types directly to `requirements-types.txt` so that they are installed and cached as part of the Github actions install dependencies step and not the `type-check` step.

This should slightly speed up the Github action and keep the `type-check` step output cleaner.


Not sure exactly why but the CI started complaining about a couple of things after I made the req file changes which is why there are some additional changes in the PR.

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

Kilo59 committed:
add typing stubs

ghost commented:
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=8501&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

####

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/great-expectations/great_expectations/8501/8f9c8876/bff007df403e57debc15713b990d4d8dc8319483.svg)](https://app.codesee.io/r/reviews?pr=8501&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
<img src="https://app.codesee.io/Legend_v3.png" alt="CodeSee Map legend" width="220" height="156">
</details>

Kilo59 committed:
remove `python-version` cli command being used by default

This is also specified in the `pyproject.toml` file.
Having both is bound to lead to confusion

Kilo59 committed:
Merge branch 'develop' into m/_/add-type-stubs-to-reqs-file

Kilo59 committed:
Merge branch 'develop' into m/_/add-type-stubs-to-reqs-file

Kilo59 committed:
remove ignores

Kilo59 committed:
update annotation for `Store._key_class`

Kilo59 committed:
Merge branch 'develop' into m/_/add-type-stubs-to-reqs-file

Kilo59 committed:
last type errors

Kilo59 committed:
Merge remote-tracking branch 'origin/m/_/add-type-stubs-to-reqs-file' into m/_/add-type-stubs-to-reqs-file

tyler-hoffman reviewed (approved):
Seems reasonable to me
